modules/c++/math.linear/include/math/linear/MatrixMxN.h

```diff
@@ -43,10 +43,30 @@ template<typename _T> inline bool equals(const _T& e1, const _T& e2, _T eps)
 
 template<> inline bool equals(const float& e1, const float& e2)
 {
+    if (e1 == std::numeric_limits<float>::infinity() &&
```

Comment:
Fixed, and I’ve confirmed that equality for numeric_limits<float>::infinity() works like that on both Linux and Windows.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Wednesday, April 13, 2016 9:51 AM
To: mdaus/coda-oss <coda-oss@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>
Subject: Re: [mdaus/coda-oss] Inf equality (#101)


In modules/c++/math.linear/include/math/linear/MatrixMxN.h<http://cp.mcafee.com/d/5fHCNAe3x0SyMVtZZZxxUTsSOMOe76zBZYS-epophd79EVdETuKYUqerCzBUSyyOyC-qen67Pozm6P9X3BvS50kGxHtendGEqTjBPoUxX7T7fZvAm67DTQTeLsKCCMPOoUQsTKU-eWyaqRQRrTsVkffGhBrwqrhdFCXYyUqenAkmkXL6XCOsVHkiP2cFASO7bUGxHtcJl3iZqkNpIY_4rEwoxaBCWkBm0H5YQsLLTsppdAaJMJZ00Lcfc8IBSm4tbydorcDIel_okelgRKDbwGx4jUzapdIxOVJYsCMr1vF6y0QJGEqTjblgQLmB3iWq80S3FEwYmrQ-kONEwd5068Thd59JyVL57GCFBi>:

> @@ -43,10 +43,30 @@ template<typename _T> inline bool equals(const _T& e1, const _T& e2, _T eps)

>

>  template<> inline bool equals(const float& e1, const float& e2)

>  {

> +    if (e1 == std::numeric_limits<float>::infinity() &&

Rather than this, shouldn't the base equals() method that takes in an eps be updated to do this?

To confirm, if you say

float e1 = std::numeric_limits<float>::infinity();

float e2 = std::numeric_limits<float>::infinity();

bool val =  e1 == e2;

val will be true right? If so, I agree that we should add this logic.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/5fHCNEqdEIenvvvooudTdIIczxNEVvvdLzCm6kjhOqejqdTHLe6zCVEVudEEIEFLCzBNxYS8RxIOuMVnZxg5aEqTjBPqG6JQVsSe8uNZNP_nV5xxVZZdPHTbFFIcYCed7dXKfzKEyCJtdmZTel3PWApmU6CQPqpK_8K6zBV55BeXNKVIDeqR4IMzapdIxO-aEqTjblgQLmBcmrffN6WeeOf-Cqeupv8cEIEK47D4m23HG9EVsphuvKCqejhOMeuR0wNZy06hvd7bXZT6mjp2Hsbvg0bP3P2b9tBx7iUzm6P9X3BvS53BkdrFOUaEh4-8OCjr8sKrv79I6MnWhEwdbqG6JQORkdbRFgQKCy0dwWq8f5CZfBcIq83hg1ydQjhiroKryb2NQQIO>


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.